src/handlers.ts

```diff
@@ -181,128 +184,159 @@ const ReplyActionSchema = z.object({
+        ];
+
+        const conversation =
+            objectToReplyTo.replyTargetId || objectToReplyTo.id!;
+        const mentions = [
+            new Mention({
+                href: attributionActor.id,
+                name: attributionActor.name,
+            }),
+        ];
+
+        const replyId = apCtx.getObjectUri(Note, {
+            id: uuidv4(),
         });
-    }
 
-    const to = PUBLIC_COLLECTION;
-    const cc = [attributionActor, apCtx.getFollowersUri(ACTOR_DEFAULT_HANDLE)];
+        const reply = new Note({
+            id: replyId,
+            attribution: actor,
+            replyTarget: objectToReplyTo,
+            content: data.content,
+            summary: null,
+            published: Temporal.Now.instant(),
+            contexts: [conversation],
+            tags: mentions,
+            to: to,
+            ccs: cc,
+        });
 
-    const conversation = objectToReplyTo.replyTargetId || objectToReplyTo.id!;
-    const mentions = [
-        new Mention({
-            href: attributionActor.id,
-            name: attributionActor.name,
-        }),
-    ];
+        const createId = apCtx.getObjectUri(Create, {
+            id: uuidv4(),
+        });
 
-    const replyId = apCtx.getObjectUri(Note, {
-        id: uuidv4(),
-    });
+        const create = new Create({
+            id: createId,
+            actor: actor,
+            object: reply,
+            to: to,
+            ccs: cc,
+        });
 
-    const reply = new Note({
-        id: replyId,
-        attribution: actor,
-        replyTarget: objectToReplyTo,
-        content: data.content,
-        summary: null,
-        published: Temporal.Now.instant(),
-        contexts: [conversation],
-        tags: mentions,
-        to: to,
-        ccs: cc,
-    });
+        const activityJson = await create.toJsonLd();
 
-    const createId = apCtx.getObjectUri(Create, {
-        id: uuidv4(),
-    });
+        await ctx.get('globaldb').set([create.id!.href], activityJson);
+        await ctx.get('globaldb').set([reply.id!.href], await reply.toJsonLd());
 
-    const create = new Create({
-        id: createId,
-        actor: actor,
-        object: reply,
-        to: to,
-        ccs: cc,
-    });
+        await addToList(ctx.get('db'), ['outbox'], create.id!.href);
 
-    const activityJson = await create.toJsonLd();
+        const account = await accountService.getByApId(actor?.id!);
 
-    await ctx.get('globaldb').set([create.id!.href], activityJson);
-    await ctx.get('globaldb').set([reply.id!.href], await reply.toJsonLd());
+        const parentPost = await postService.getByApId(objectToReplyTo.id!);
 
-    await addToList(ctx.get('db'), ['outbox'], create.id!.href);
+        const replyData = {
+            content: data.content,
+            type: PostType.Note,
+            audience: Audience.Public,
+            apId: replyId,
+            inReplyTo: parentPost,
+            threadRoot:
```

Comment:
The way I have implemented this is:-
1. ThreadRoot for a root post is NULL.
2. ThreadRoot for any reply is the ThreadRoot of it's parent. If the ThreadRoot of parent is NULL, parent is the root.

Reply:
Nice, I like it